actors/market/src/lib.rs

```diff
@@ -810,38 +790,17 @@ impl Actor {
                         })?;
                     }
 
-                    let (slash_amount, remove_deal) =
-                        st.process_deal_update(rt.store(), &state, &deal, curr_epoch)?;
-
-                    if slash_amount.is_negative() {
-                        return Err(actor_error!(
-                            illegal_state,
-                            format!(
-                                "computed negative slash amount {} for deal {}",
-                                slash_amount, deal_id
-                            )
-                        ));
-                    }
+                    let (slash_amount, remove_deal) = st.process_deal_update(
```

Comment:
If we process terminated deals synchronously, there will never be any slashing here and the only removal will be of completed deals. Inside process_deal_update, it will become an error to ever have non-empty slash epoch. (A future migration can remove it).

Reply:
I was going to say my comment above is still valid, so maybe we could assert slash_amount is zero. But perhaps again the case is a deal marked for termination before this code goes live. Please add comments (and tests) describing this, so we can simplify it later.